### PR TITLE
Cleanup tests in ServerBuilderTest.

### PR DESCRIPTION
**Pull request by alexander-poulikakos**
## Change list

Refactored unit tests.
 
## Types of changes

What types of changes are you proposing/introducing to Java client?
_Put an `x` in the boxes that apply_

- [x] No changes in production code
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Details

Cleanup unit tests.

### CODE REVIEW

**Comment by alexander-poulikakos** [comment]
In the unit tests that changes the URL (`checkAbilityToStartService*UsingFlag`) I chnaged so the Assertion of URL is dome before starting the service. This makes it fail faster, when failing. No need to wait for client to timeout.

Moved duplicated try/catch clauses to @After method to make code cleaner.

File: src/test/java/io/appium/java_client/localserver/ServerBuilderTest.java [hunk 1]
```diff
@@ -1,434 +1,311 @@
...
+            if (testIP != null) {
+                break;
+            }
+        }
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        if (service != null) {
+            service.stop();
```
Comment (TikhomirovSergey) [inline src/test/java/io/appium/java_client/localserver/ServerBuilderTest.java]: @alexander-poulikakos 
Please don't forget to 
```
service = null;
```

File: src/test/java/io/appium/java_client/localserver/ServerBuilderTest.java [hunk 2]
```diff
@@ -1,434 +1,311 @@
...
+        }
+        if (stream != null) {
+            stream.close();
```
Comment (TikhomirovSergey) [inline src/test/java/io/appium/java_client/localserver/ServerBuilderTest.java]: Please don't forget to 
```
stream = null;
```

File: src/test/java/io/appium/java_client/localserver/ServerBuilderTest.java [hunk 3]
```diff
@@ -1,434 +1,311 @@
...
+        }
+        if (file != null && file.exists()) {
+            file.delete();
```
Comment (TikhomirovSergey) [inline src/test/java/io/appium/java_client/localserver/ServerBuilderTest.java]: Please don't forget to 
```
file = null;
```

Comment (alexander-poulikakos) [inline src/test/java/io/appium/java_client/localserver/ServerBuilderTest.java]: No need to se values to null in @After method. JUnit will make a new instance of ServerBuilderTest for each @Test which will automatically reset all member fields.

File: src/test/java/io/appium/java_client/localserver/ServerBuilderTest.java (hunk 2 above)
Comment (alexander-poulikakos) [inline src/test/java/io/appium/java_client/localserver/ServerBuilderTest.java]: No need to se values to null in @After method. JUnit will make a new instance of ServerBuilderTest for each @Test which will automatically reset all member fields.

File: src/test/java/io/appium/java_client/localserver/ServerBuilderTest.java (hunk 1 above)
Comment (alexander-poulikakos) [inline src/test/java/io/appium/java_client/localserver/ServerBuilderTest.java]: No need to se values to null in @After method. JUnit will make a new instance of ServerBuilderTest for each @Test which will automatically reset all member fields.

**Pull request merged**